File: baekjoon/String/1764.cpp
// 백준 1764번 : 듣보잡
#include <iostream>
#include <map>

using namespace std;

int main() {
    int N, M, ans = 0;
    map<string, int> m;
    string name = "";

    cin >> N >> M;
    while (N--) {
        cin >> name;
        m[name]++;
    }
    while (M--) {
        cin >> name;
        m[name]++;
        if (m[name] >= 2) {
            ans++;
        }
    }
    cout << ans << '\n';
    for (auto it : m) {
        if (it.second >= 2) {
            cout << it.first << '\n';
        }
    }
    return 0;
}

File: baekjoon/String/1764.py
# 백준 1764번 : 듣보잡
if __name__ == "__main__":
    N, M = map(int, input().split())

    st1 = set()
    st2 = set()

    for _ in range(N):
        st1.add(input())
    
    for _ in range(M):
        st2.add(input())

    ans = sorted(list(st1 & st2))
    print(len(ans))

    for i in ans:
        print(i)